### PR TITLE
update web3 data changed callback

### PR DESCRIPTION
**Pull request by James-Sangalli**
https://github.com/AlphaWallet/alpha-wallet-ios/pull/2161

### CODE REVIEW

**Review by hboon — CHANGES_REQUESTED** [review]
Also needs changes like this https://github.com/AlphaWallet/alpha-wallet-ios/pull/2161/files#diff-21eeed1f14eca84e42fe18b15a229ae9L67

**Comment by James-Sangalli** [comment]
> Also needs changes like this https://github.com/AlphaWallet/alpha-wallet-ios/pull/2161/files#diff-21eeed1f14eca84e42fe18b15a229ae9L67

Thanks @hboon, updated.

File: ERCs/ERC20/layouts/item-view-approvalObtained.en.js [hunk 1]
```diff
@@ -17,15 +17,16 @@ class Token {
...
-            <span><bold><h3>Approved to move ${this.props.amount} ${this.props.symbol} from ${this.props.from}</h3></bold></span>
+            <span><bold><h3>Approved to move ${this.props.card.amount} ${this.props.token.symbol} from ${this.props.from}</h3></bold></span>
```
Comment (hboon) [inline ERCs/ERC20/layouts/item-view-approvalObtained.en.js:20]: `this.props.from` should be `this.props.card.from`

File: ERCs/ERC20/layouts/item-view-ownerApproved.en.js [hunk 2]
```diff
@@ -17,15 +17,16 @@ class Token {
...
-            <span><bold><h3>Gave approval to move ${this.props.amount} ${this.props.symbol} to ${this.props.to}</h3></bold></span>
+            <span><bold><h3>Gave approval to move ${this.props.card.amount} ${this.props.token.symbol} to ${this.props.to}</h3></bold></span>
```
Comment (hboon) [inline ERCs/ERC20/layouts/item-view-ownerApproved.en.js:20]: `this.props.to` should be `this.props.card.to`

File: ERCs/ERC20/layouts/receivedApproval.en.js [hunk 3]
```diff
@@ -1,18 +1,18 @@
...
             <span><bold><h3>${message}</h3></bold></span>
-            <p>Approved to move ${this.props.amount} ${this.props.symbol} from ${this.props.from}</p>
+            <p>Approved to move ${this.props.card.amount} ${this.props.token.symbol} from ${this.props.from}</p>
```
Comment (hboon) [inline ERCs/ERC20/layouts/receivedApproval.en.js:15]: `this.props.from` should be `this.props.card.from`

File: ERCs/ERC721/layouts/item-view-sent.en.js [hunk 4]
```diff
@@ -18,7 +18,7 @@ class Token {
...
-            <span><bold><h3>Sent ${this.props.symbol} tokenId: ${this.props.tokenId}</h3></bold></span>
+            <span><bold><h3>Sent ${this.props.token.symbol} tokenId: ${this.props.tokenId}</h3></bold></span>
```
Comment (hboon) [inline ERCs/ERC721/layouts/item-view-sent.en.js:21]: `this.props.tokenId` should be `this.props.token.tokenId`

File: ERCs/ERC721/layouts/receivedApproval.en.js [hunk 5]
```diff
@@ -1,18 +1,18 @@
...
             <span><bold><h3>${message}</h3></bold></span>
-            <p>Approved to move tokenId: ${this.props.tokenId} ${this.props.symbol} from ${this.props.from}</p>
+            <p>Approved to move tokenId: ${this.props.token.tokenId} ${this.props.token.symbol} from ${this.props.from}</p>
```
Comment (hboon) [inline ERCs/ERC721/layouts/receivedApproval.en.js:15]: `this.props.from` should be `this.props.card.from`

**Review by hboon — CHANGES_REQUESTED** [review]
A few more. I just did a quick pass. See if they are right.

**Comment by James-Sangalli** [comment]
@hboon sorry, should be good now!

**Review by hboon — APPROVED** [review]

**Comment by hboon** [comment]
Good to merge. Thanks!

**Pull request merged**